contrib/src/main/java/akka/stream/contrib/FileTailSource.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib;
+
+import akka.NotUsed;
+import akka.stream.Attributes;
+import akka.stream.Outlet;
+import akka.stream.SourceShape;
+import akka.stream.javadsl.Source;
+import akka.stream.stage.*;
+import akka.util.ByteString;
+import scala.concurrent.duration.FiniteDuration;
+import scala.util.Failure;
+import scala.util.Success;
+import scala.util.Try;
+
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.nio.channels.CompletionHandler;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+/**
+ * Read the entire contents of a file, and then when the end is reached, keep reading
+ * newly appended data. Like the unix command `tail -f`.
+ *
+ * Aborting the stage can be done by combining with a [[akka.stream.KillSwitch]]
```

Comment:
Hmmm. I still lean towards simpler is better here. Especially as you can compose to achieve the same behavior.